### ISSUE TITLE
Incorrect warning on INFO: Invalid Type for attribute: _FillValue <class 'numpy.float32'>

### ISSUE THREAD

**Issue by mikeggrant-eumetsat** [cedadev/cf-checker#87]
Using the current master branch on a netcdf like:

```
netcdf my_lovely_netcdf {
dimensions:
        y = UNLIMITED ; // (614 currently)
        x = 28 ;
variables:
        float latitude(y, x) ;
                latitude:_FillValue = -999.f ;
                latitude:standard_name = "latitude" ;
                latitude:long_name = "latitude" ;
                latitude:units = "degree_north" ;
                latitude:description = "Latitude for each pixel in every scanline." ;
}
```

I'm seeing an info message like:

```
------------------
Checking variable: latitude
------------------
Attribute _FillValue
INFO: Invalid Type for attribute: _FillValue <class 'numpy.float32'>

```

I think the issue is that the check whether an attribute is numeric or not is not working.  On line 2088 of src/cfchecker/cfchecks.py:
`            elif numpy.issubdtype(attrType, numpy.int) or numpy.issubdtype(attrType, numpy.float):`
the "numpy.float" in the subdtype check seems not to recognise the float32 returned by netCDF4/numpy as a float, possibly because this type resolves to a python float.

Looking at https://numpy.org/doc/stable/reference/arrays.scalars.html, I amended the types to be the parents of the integer and float hierarchies in 
`            elif numpy.issubdtype(attrType, numpy.integer) or numpy.issubdtype(attrType, numpy.floating):`

This makes the info message disappear.

I'm not 100% sure if this is the right fix, as it may bring in other unwanted types (e.g. long floats) and still excludes others (complex numbers).

Incidentally, I'm using netcdf4 1.5.4 and numpy 1.19.1 fresh off conda-forge; possibly this worked in the past and something has changed (there's a lot of google hits on "numpy.float deprecated", but these may be related to a different issue).

**Comment by minxu74** [comment]
@mikeggrant-eumetsat I found that there was no that warning on INFO message if the version of python netCDF4 is 1.5.3. But the warning appeared when the netCD4 version was 1.5.4.  So something has changed in the upstream netCDF4 library

**Comment by mikeggrant-eumetsat** [comment]
@minxu74 thanks for checking that!  Looking at the [netcdf4-python Changelog](https://github.com/Unidata/netcdf4-python/blob/master/Changelog), I note ` * bump minimal numpy version to 1.9 (first version to have tobytes()).` as the only suspicious entry.  Was your 1.5.3 also with a recent numpy?

**Comment by minxu74** [comment]
@mikeggrant-eumetsat  the netcdf/1.5.3 is with numpy 1.18.1 and the netcdf/1.5.4 is with numpy 1.19.1.  After updating the numpy/1.18.1 to numpy 1.19.1 with netcdf/1.5.3, the same warning appeared. So it seemed that the warning is originated from numpy.

**Comment by mikeggrant-eumetsat** [comment]
Thanks again for the followup :)  That perhaps aligns with the numpy float changes I hinted at.  Those were:
https://numpy.org/doc/stable/release/1.19.0-notes.html#issubdtype-no-longer-interprets-float-as-np-floating
```numpy.issubdtype had a FutureWarning since NumPy 1.14 which has expired now. This means that certain input where the second argument was neither a datatype nor a NumPy scalar type (such as a string or a python type like int or float) will now be consistent with passing in np.dtype(arg2).type. This makes the result consistent with expectations and leads to a false result in some cases which previously returned true.```

I haven't followed it up in enough detail to be sure, but it looks quite likely.  The original code has `numpy.float`, which resolves to the python 'float' type, so I'd tentatively pin the blame to it until someone checks properly!  If so, it would support the suggested change above though.

**Comment by zklaus** [comment]
I think @mikeggrant-eumetsat is right on the money here and @TUBklima's patch seems appropriate. @RosalynHatcher, @philipkershaw, do you need anything else to make this happen?

**Comment by RosalynHatcher** [comment]
@zklaus: Sorry I dropped the ball here. :-( Don't think I need anything else.  I'll endeavour to get it done later this week.

**Comment by zklaus** [comment]
@RosalynHatcher no worries, been there, done that ;) just a friendly prod.

**Comment by zklaus** [comment]
Thanks @RosalynHatcher !
